### PR TITLE
Create Renderless extra param component

### PR DESCRIPTION
**Pull request by pmareke**
EX-4605

### CODE REVIEW

**Review by davidmfempathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/extra-params/components/__tests__/renderless-extra-params.spec.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    expect(extraParamsProvidedCallback).toHaveBeenCalledTimes(0);
+  });
+
+  it('emits UserChangedExtraParams event when the update method is called', () => {
+    const userChangedExtraParamsCallback = jest.fn();
+    const { wrapper } = renderRenderlessExtraParams({
+      extraParamName: 'warehouse',
+      scopedSlots: {
+        default: `
+          <template #search="{ defaultValue, updateValue }">
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/__tests__/renderless-extra-params.spec.ts:99]: I don't know how this is working, you are defining a scoped slot that has a scoped slot named `search` 🤣,  For me it is far more clear if we just use the template like in other tests. Here I have exactly the input of the test, properly defined, a template with a default scoped slot.

```ts
    const { wrapper } = renderRenderlessExtraParams({
      extraParamName: 'warehouse',
      template: `
        <RenderlessExtraParam #default="{ defaultValue, updateValue }">
          <button data-test="custom-slot" @click="updateValue(45678)">Update warehouse</button>
        </RenderlessExtraParam>`
    });

```

 Moreover, wasn't that scoped slots property giving us some troubles?

```
    const { wrapper } = renderRenderlessExtraParams({
      extraParamName: 'warehouse',
      template: `
        <RenderlessExtraParam #default="{ defaultValue, updateValue }">
          <button data-test="custom-slot" @click="updateValue(45678)">Update warehouse</button>
        </RenderlessExtraParam>`
    });
```

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/extra-params/components/renderless-extra-param.vue [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    /**
+     * The extra param's default value.
+     *
+     * @public
+     */
+    @Prop()
+    public defaultValue?: unknown;
+
+    @Watch('defaultValue')
+    syncUpdateValue(newValue: unknown): void {
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/extra-params/components/renderless-extra-param.vue:46]: It'll change if you change the prop sent to the component right?

File: packages/x-components/src/x-modules/extra-params/components/__tests__/extra-params.spec.ts [hunk 3]
```diff
@@ -38,19 +38,23 @@ describe('testing extra params component', () => {
...
+      }
+    );
 
     await wrapper.setProps({ values: { warehouse: 5678 } });
 
-    expect(extraParamsProvidedCallback).toHaveBeenCalledWith<[WirePayload<Dictionary<unknown>>]>({
-      eventPayload: { warehouse: 5678 },
-      metadata: { moduleName: 'extraParams' }
-    });
-    expect(extraParamsProvidedCallback).toHaveBeenCalledTimes(2);
```
[the remark this answers is not in the document]
Comment (pmareke) [reply packages/x-components/src/x-modules/extra-params/components/__tests__/extra-params.spec.ts:53]: My fault, I asume that `toHaveBeenNthCalledWith` was the combination of `toHaveBeenCalledWith` and `toHaveBeenCalledTimes`, but it's not.

File: packages/x-components/src/x-modules/extra-params/components/__tests__/renderless-extra-params.spec.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    expect(extraParamsProvidedCallback).toHaveBeenCalledWith<[WirePayload<Dictionary<unknown>>]>({
+      eventPayload: { warehouse: 1234 },
+      metadata: { moduleName: 'extraParams' }
+    });
+
+    expect(extraParamsProvidedCallback).toHaveBeenCalledTimes(1);
+  });
+
+  // eslint-disable-next-line max-len
+  it("not emits ExtraParamsProvided event when the component receives a default value if it's in the store", () => {
```
Comment (davidmfempathy) [inline packages/x-components/src/x-modules/extra-params/components/__tests__/renderless-extra-params.spec.ts:72]: doesn't emit

**Review by guillei10 — DISMISSED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Review by guillei10 — APPROVED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Pull request merged**